modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/ObjectEntryModelListener.java

```diff
@@ -80,48 +88,79 @@ private void _executeObjectActions(
 				objectEntry.getModelClassName(), objectEntry.getCompanyId(),
 				objectActionTriggerKey,
 				_getPayloadJSONObject(
-					objectActionTriggerKey, originalObjectEntry, objectEntry),
+					objectActionTriggerKey, originalObjectEntry, objectEntry,
+					userId),
 				userId);
 		}
 		catch (PortalException portalException) {
 			throw new ModelListenerException(portalException);
 		}
 	}
 
+	private String _getExternalModel(ObjectEntry objectEntry, long userId)
+		throws PortalException {
+
+		User user = _userLocalService.getUser(userId);
+
+		DefaultDTOConverterContext defaultDTOConverterContext =
+			new DefaultDTOConverterContext(
+				false, Collections.emptyMap(), _dtoConverterRegistry, null,
+				user.getLocale(), null, user);
+
+		DTOConverter<ObjectEntry, ?> dtoConverter =
+			(DTOConverter<ObjectEntry, ?>)_dtoConverterRegistry.getDTOConverter(
+				ObjectEntry.class.getName());
+
+		try {
+			Object externalModel = dtoConverter.toDTO(
+				defaultDTOConverterContext, objectEntry);
+
+			return externalModel.toString();
+		}
+		catch (Exception exception) {
+			_log.error(exception, exception);
+		}
+
+		return objectEntry.toString();
+	}
+
 	private JSONObject _getPayloadJSONObject(
 			String objectActionTriggerKey, ObjectEntry originalObjectEntry,
-			ObjectEntry objectEntry)
-		throws JSONException {
+			ObjectEntry objectEntry, long userId)
+		throws PortalException {
```

Comment:
We can really throw a PortalException? or any kind of exception if the DTOConverter fails?

Reply:
I think so, because the DTOConverter's methods throw Exception. 
But this change does not come from the DTO but from the getUser method that I use in the _getExternalModel method to get the user. Because that method throws a PortalException